src/askui/models/openrouter/handler.py

```diff
@@ -0,0 +1,77 @@
+import os
+from typing import Type
+
+from openai import OpenAI
+from typing_extensions import override
+
+from askui.models.exceptions import QueryNoResponseError
+from askui.models.models import GetModel
+from askui.models.types.response_schemas import ResponseSchema
+from askui.utils.image_utils import ImageSource
+
+from .prompts import PROMPT_QA
+from .settings import OpenRouterSettings
+
+
+class OpenRouterGetModel(GetModel):
+    def __init__(self, settings: OpenRouterSettings):
+        self._settings = settings
+
+        _open_router_key = os.getenv("OPEN_ROUTER_API_KEY")
+        if _open_router_key is None:
+            error_msg = "OPEN_ROUTER_API_KEY is not set"
+            raise ValueError(error_msg)
```

Comment:
I would add this to the `OpenRouterSettings` as a property and let `OpenRouterSettings` extend `BaseSettings`. Every property not initialised by the consumer through the constructor is automatically initialized from the env. Is more consistent and allows for more powerful, more declarative validation from my perspective and ensures environment variables are consistently prefixed. Also more powerful as users can actually set then all settings through environment or programmatically.